common/eigenda_network.go

```diff
@@ -61,11 +70,14 @@ func (n EigenDANetwork) String() string {
 	return string(n)
 }
 
+// TODO: Consider deriving network from chain ID via eth_getChainId or similar rpc call
+//
+//	vs hardcoding the network names and forcing user specification
```

Comment:
I think its prob best to let users specify chainID but then crash if specified chainID doesnt match the rpc_url. Prevent someone from dispersing to mainnet for a while when they really wanted a testnet proxy